contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,516 @@
+          values: []
+          effects: []
+        action: allowed
+```
+
+
+
+### Priority classes
+
+Priority class usage can be regulated using fine-grain rules with `priorityClasseName` field.
+
+##### required
+
+this example requires a `priorityClass` that is either `high-priority` or `critical-job`
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    priorityClasseNames:
+      - match: ["high-priority","critical-job"]
+        action: required
+```
+
+
+
+##### Allow
+
+In this example, we only allow the `critical-job` priority.
+
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    priorityClasseNames:
+      - match: ["critical-priority"]
+        action: allowed
+```
+
+
+
+
+### Node Selector
+
+The `nodeSelector` field makes it possible to specify what nodeSelectors are required, allowed and denied. As for other components, `required` nodeSelectors are automatically considered as allowed.
+
+#### Examples
+
+##### Complete policy
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    nodeSelectors:
+      - match:
+        - beta.kubernetes.io/arch: ["amd64", "arm64"]
+        - team: []
+        action: required
+      - match:
+        - failure-domain.beta.kubernetes.io/region: []
+        - disk: ["ssd", "hdd"]
+        action: allowed
+
+```
+
+
+In this example, pods can be scheduled if they match the two following:
+-  a nodeSelector `beta.kubernetes.io/arch=amd64` or `beta.kubernetes.io/arch=arm64`.
+- a nodeSelector with the key `team`.
+
+They can also optionally specify:
+-  `disk: ssd` nodeSelector,
+-  `disk: hdd` nodeSelector,
+-  `failure-domain.beta.kubernetes.io/region` nodeSelector with any value.
+
+
+### Affinity rules
+
+As anti-affinity rules are really time-consuming, we must be able to restrict their usage with `allowedAffinities`.
+`allowedAffinities` is supposed to keep a coarse-grained approach in allowing affinities. For each type (`nodeAffinities`, `podAffinities`, `podAntiAffinities`) a schedulingpolicy can list allowed constraints (`requiredDuringSchedulingIgnoredDuringExecution`
+or `requiredDuringSchedulingIgnoredDuringExecution`).
+
+If `allowedAffinities` is totally absent from the spec, no affinity is allowed whatever its kind.
+
+#### Examples
+
+##### Basic policy
+
+
+
+```yaml
+apiVersion: policy/v1alpha1
+kind: SchedulingPolicy
+metadata:
+  name: mySchedulingPolicy
+spec:
+  bindingMode: all
+  namespaces:
+    - default
+  rules:
+    nodeAffinities:
+      - match:
+        - keys: ["failure-domain.beta.kubernetes.io/region","authorized-region"]
+          operators: ["In","NotIn"]
+          values: ["eu-2", "us-1"]
```

Comment:
This is confusing as I commented above. I'd rather break these into two matches.